packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -27,67 +33,161 @@ import {
   getTruncateNumbersWithPound,
   getSettings,
   saveSettings as saveSettingsAction,
+  RootState,
+  WorkspaceSettings,
 } from '@deephaven/redux';
-import { DbNameValidator, TimeUtils } from '@deephaven/utils';
+import { assertNotNull, DbNameValidator, TimeUtils } from '@deephaven/utils';
 import './FormattingSectionContent.scss';
+import type { DebouncedFunc } from 'lodash';
 
 const log = Log.module('FormattingSectionContent');
 
-export class FormattingSectionContent extends PureComponent {
+type FormatterItem = {
+  columnType: string;
+  columnName: string;
+  format: Partial<TableColumnFormat>;
+  id?: number;
+  isNewRule?: boolean;
+};
+
+type FormatOption = {
+  defaultFormatString?: string;
+};
+
+interface FormattingSectionContentProps {
+  formatter?: FormatterItem[];
+  defaultDateTimeFormat?: string;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  timeZone?: string;
+  truncateNumbersWithPound?: boolean;
+  settings?: WorkspaceSettings;
+  saveSettings?: (settings: WorkspaceSettings) => void;
+  scrollTo: (x: number, y: number) => void;
+  defaultDecimalFormatOptions?: FormatOption;
+  defaultIntegerFormatOptions?: FormatOption;
+  defaults: {
+    defaultDateTimeFormat: string;
+    defaultDecimalFormatOptions: FormatOption;
+    defaultIntegerFormatOptions: FormatOption;
+    showTimeZone: boolean;
+    showTSeparator: boolean;
+    timeZone: string;
+  };
+}
+
+interface FormattingSectionContentState {
+  formatSettings?: FormatterItem[];
+  formatRulesChanged: boolean;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  timeZone?: string;
+  defaultDateTimeFormat?: string;
+  defaultDecimalFormatOptions?: FormatOption;
+  defaultIntegerFormatOptions?: FormatOption;
+  truncateNumbersWithPound?: boolean;
+  timestampAtMenuOpen: Date;
+}
+
+export class FormattingSectionContent extends PureComponent<
+  FormattingSectionContentProps,
+  FormattingSectionContentState
+> {
+  static defaultProps = {
+    scrollTo: (): void => undefined,
+    defaults: {
+      defaultDateTimeFormat:
+        DateTimeColumnFormatter.DEFAULT_DATETIME_FORMAT_STRING,
+      defaultDecimalFormatOptions: {
+        defaultFormatString: DecimalColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      defaultIntegerFormatOptions: {
+        defaultFormatString: IntegerColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      showTimeZone: false,
+      showTSeparator: true,
+      timeZone: DateTimeColumnFormatter.DEFAULT_TIME_ZONE_ID,
+    },
+  };
+
   static inputDebounceTime = 250;
 
-  static focusFirstInputInContainer(container) {
-    const input = container.querySelector('input, select, textarea');
+  static focusFirstInputInContainer(container: HTMLElement | null): void {
+    const input = container?.querySelector('input, select, textarea');
     if (input) {
-      input.focus();
+      (input as HTMLElement).focus();
```

Comment:
Prefer to type on the declaration instead of casting after:
```
const input: HTMLElement | null | undefined = container?.querySelector(
      'input, select, textarea'
    );
```